macros/src/main/scala/pureconfig/Derivation.scala

```diff
@@ -48,17 +48,44 @@ class DerivationMacros(val c: whitebox.Context) extends LazyContextParser with M
     // but we may be unable to parse `Lazy` trees)
     val isHeadImplicitADerivation = c.openImplicits.headOption.exists(_.pre =:= typeOf[Derivation.type])
 
-    // check the `-Xmacro-settings:materialize-derivations` scalac flag
-    if (!isDerivationEnabled || !isHeadImplicitADerivation) {
-      // when not present, simply render the base `Derivation` constructor with `implcitly[A]`. This results in the same
-      // behavior as without `Derivation` (apart from the extra wrapper).
-      q"_root_.pureconfig.Derivation.Successful(_root_.scala.Predef.implicitly[${weakTypeOf[A]}])"
+    // check if the materialization was called explicitly, in which case we want to have the nicer compiler error
+    // messages
+    val isMaterializationExplicitCall = {
+      val firstMacroCall = c.enclosingMacros.reverse.find(ctx => ctx.prefix.tree.tpe =:= ctx.typeOf[Derivation.type])
+      firstMacroCall.fold(false)(_.openImplicits.isEmpty)
```

Comment:
`.fold(false)(f)` is equal to `.exists(f)`.

Reply:
I've improved that expression in bfc162a.